### ISSUE TITLE
--num_worker  is the GPUs numbers?

### ISSUE THREAD

**Issue by wxw420** [yjxiong/temporal-segment-networks#42]
> python tools/eval_net.py ucf101 1 rgb /home/xs/deep-learning/dataset/UCF101_FRAMES/ models/ucf101/tsn_bn_inception_rgb_deploy.prototxt models/ucf101_split_1_tsn_rgb_reference_bn_inception.caffemodel --num_worker 2 --save_scores ucf101score

I have two K80 GPU,  when I run the eval_net.py ,it warning like this:

I1221 23:06:41.762329  4542 net.cpp:294] Network initialization done.
I1221 23:06:41.762334  4542 net.cpp:295] Memory required for data: 74005652
Setting device 2
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:45.627977  4567 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 3
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:45.829156  4572 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 4
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:46.244467  4577 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 5
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:46.447710  4580 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 6
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:46.553652  4583 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 7
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:46.754379  4588 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 8
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:46.964293  4593 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 9
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:47.169553  4596 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 10
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:47.274942  4599 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 11
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:47.480448  4604 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 12
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:47.685133  4609 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 13
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:47.993522  4612 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 14
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:48.096205  4615 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 15
WARNING: Logging before InitGoogleLogging() is written to STDERR
F1221 23:06:48.302572  4620 common.cpp:187] Check failed: error == cudaSuccess (10 vs. 0)  invalid device ordinal
*** Check failure stack trace: ***
Setting device 16
Setting device 17

can you help me? thank you very much.

**Comment by yjxiong** [comment]
This means the initialization of the networks has failed. Please run with `--num_worker 1` to see what is the error.

**Comment by wxw420** [comment]
I run with --num_worker 1
wang@xs:~/deep-learning/TSN-action$ python tools/eval_net.py ucf101 1 rgb /home/xs/deep-learning/dataset/UCF101_FRAMES/ models/ucf101/tsn_bn_inception_rgb_deploy.prototxt models/ucf101_split_1_tsn_rgb_reference_bn_inception.caffemodel --num_worker 1 --save_scores score_file
Namespace(caffe_path='./lib/caffe-action/', dataset='ucf101', flow_x_prefix='flow_x_', flow_y_prefix='flow_y_', frame_path='/home/xs/deep-learning/dataset/UCF101_FRAMES/', gpus=None, modality='rgb', net_proto='models/ucf101/tsn_bn_inception_rgb_deploy.prototxt', net_weights='models/ucf101_split_1_tsn_rgb_reference_bn_inception.caffemodel', num_frame_per_video=25, num_worker=1, rgb_prefix='img_', save_scores='score_file', split=1)
ucf101
parse frames under folder /home/xs/deep-learning/dataset/UCF101_FRAMES/
0 videos parsed
200 videos parsed
400 videos parsed
600 videos parsed
800 videos parsed
1000 videos parsed
1200 videos parsed
1400 videos parsed
1600 videos parsed
1800 videos parsed
2000 videos parsed
2200 videos parsed
2400 videos parsed
2600 videos parsed
2800 videos parsed
3000 videos parsed
3200 videos parsed
3400 videos parsed
3600 videos parsed
3800 videos parsed
4000 videos parsed
4200 videos parsed
4400 videos parsed
4600 videos parsed
4800 videos parsed
5000 videos parsed
5200 videos parsed
5400 videos parsed
5600 videos parsed
5800 videos parsed
6000 videos parsed
6200 videos parsed
6400 videos parsed
6600 videos parsed
6800 videos parsed
7000 videos parsed
7200 videos parsed
7400 videos parsed
7600 videos parsed
7800 videos parsed
8000 videos parsed
8200 videos parsed
8400 videos parsed
8600 videos parsed
8800 videos parsed
9000 videos parsed
9200 videos parsed
9400 videos parsed
9600 videos parsed
9800 videos parsed
10000 videos parsed
10200 videos parsed
10400 videos parsed
10600 videos parsed
10800 videos parsed
11000 videos parsed
11200 videos parsed
11400 videos parsed
11600 videos parsed
11800 videos parsed
12000 videos parsed
12200 videos parsed
12400 videos parsed
12600 videos parsed
12800 videos parsed
13000 videos parsed
13200 videos parsed
frame folder analysis done
Setting device 0
WARNING: Logging before InitGoogleLogging() is written to STDERR
I1222 18:11:38.224102  3689 net.cpp:46] Initializing net from parameters: 
name: "BN-Inception"
input: "data"
input_dim: 1
input_dim: 3
input_dim: 224
input_dim: 224
state {
  phase: TEST
}
layer {
  name: "conv1/7x7_s2"
  type: "Convolution"
  bottom: "data"
  top: "conv1/7x7_s2"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 64
    pad: 3
    kernel_size: 7
    stride: 2
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
      value: 0.2
    }
  }
}
layer {
  name: "conv1/7x7_s2_bn"
  type: "BN"
  bottom: "conv1/7x7_s2"
  top: "conv1/7x7_s2_bn"
  param {
    lr_mult: 1
    decay_mult: 0
  }
  param {
    lr_mult: 1
    decay_mult: 0
  }
  bn_param {
    slope_filler {
      type: "constant"
      value: 1
    }
    bias_filler {
      type: "constant"
      value: 0
    }
    frozen: true
  }
}
layer {
  name: "conv1/relu_7x7"
  type: "ReLU"
  bottom: "conv1/7x7_s2_bn"
  top: "conv1/7x7_s2_bn"
}
layer {
  name: "pool1/3x3_s2"
  type: "Pooling"
  bottom: "conv1/7x7_s2_bn"
  top: "pool1/3x3_s2"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "conv2/3x3_reduce"
  type: "Convolution"
  bottom: "pool1/3x3_s2"
  top: "conv2/3x3_reduce"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 64
    kernel_size: 1
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
      value: 0.2
    }
  }
}
layer {
  name: "conv2/3x3_reduce_bn"
  type: "BN"
  bottom: "conv2/3x3_reduce"
  top: "conv2/3x3_reduce_bn"
  param {
    lr_mult: 1
    decay_mult: 0
  }
  param {
    lr_mult: 1
    decay_mult: 0
  }
  bn_param {
    slope_filler {
      type: "constant"
      value: 1
    }
    bias_filler {
      type: "constant"
      value: 0
    }
    frozen: true
  }
}
layer {
  name: "conv2/relu_3x3_reduce"
  type: "ReLU"
  bottom: "conv2/3x3_reduce_bn"
  top: "conv2/3x3_reduce_bn"
}
layer {
  name: "conv2/3x3"
  type: "Convolution"
  bottom: "conv2/3x3_reduce_bn"
  top: "conv2/3x3"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 192
    pad: 1
    kernel_size: 3
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
      value: 0.2
    }
  }
}
layer {
  name: "conv2/3x3_bn"
  type: "BN"
  bottom: "conv2/3x3"
  top: "conv2/3x3_bn"
  param {
    lr_mult: 1
    decay_mult: 0
  }
  param {
    lr_mult: 1
    decay_mult: 0
  }
  bn_param {
    slope_filler {
      type: "constant"
      value: 1
    }
    bias_filler {
      type: "constant"
      value: 0
    }
    frozen: true
  }
}
layer {
  name: "conv2/relu_3x3"
  type: "ReLU"
  bottom: "conv2/3x3_bn"
  top: "conv2/3x3_bn"
}
layer {
  name: "pool2/3x3_s2"
  type: "Pooling"
  bottom: "conv2/3x3_bn"
  top: "pool2/3x3_s2"
  pooling_param {
    pool: MAX
    kernel_size: 3
    stride: 2
  }
}
layer {
  name: "inception_3a/1x1"
  type: "Convolution"
  bottom: "pool2/3x3_s2"
  top: "inception_3a/1x1"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 64
    kernel_size: 1
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
      value: 0.2
    }
  }
}
layer {
  name: "inception_3a/1x1_bn"
  type: "BN"
  bottom: "inception_3a/1x1"
  top: "inception_3a/1x1_bn"
  param {
    lr_mult: 1
    decay_mult: 0
  }
  param {
    lr_mult: 1
    decay_mult: 0
  }
  bn_param {
    slope_filler {
      type: "constant"
      value: 1
    }
    bias_filler {
      type: "constant"
      value: 0
    }
    frozen: true
  }
}
layer {
  name: "inception_3a/relu_1x1"
  type: "ReLU"
  bottom: "inception_3a/1x1_bn"
  top: "inception_3a/1x1_bn"
}
layer {
  name: "inception_3a/3x3_reduce"
  type: "Convolution"
  bottom: "pool2/3x3_s2"
  top: "inception_3a/3x3_reduce"
  param {
    lr_mult: 1
    decay_mult: 1
  }
  param {
    lr_mult: 2
    decay_mult: 0
  }
  convolution_param {
    num_output: 64
    kernel_size: 1
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
      value: 0.2
    }
  }
}
layer {
  name: "inception_3a/3x3_reduce_bn"
  type: "BN"
  bottom: "inception_3a/3x3_reduce"
  top: "inception_3a/3x3_reduce_bn"
  param {
    lr_mult: 1
    decay_mult: 0
  }
...
...
I1222 18:11:41.756407  3689 net.cpp:531] Collecting Learning Rate and Weight Decay.
I1222 18:11:41.756465  3689 net.cpp:294] Network initialization done.
I1222 18:11:41.756479  3689 net.cpp:295] Memory required for data: 74005652
段错误 (核心已转储)

非常奇怪，我装好之后，无论--num_worker  1 或者 2 ,一直提示这样的错误，但是，我期间有两次竟然成功的运行了，我并没有改动任何地方

**Comment by yjxiong** [comment]
In this case, it is probably related to your environments, like library versions and packages installed.

**Comment by wxw420** [comment]
训练网络时，用rgb输入时可以训练；用flow时又会报错…我需要重新配置一下所有库吗……

**Comment by wxw420** [comment]
非常感谢，我已经找到问题了，我拷贝的cv2.so出了问题，调整后可以正常运行了。再一次，非常感谢您分享代码

**Comment by yjxiong** [comment]
Closing this. Please feel free to reopen it if you meet any further problem.
